src/test/java/se/liu/ida/hefquin/engine/queryplan/executable/impl/ops/ExecOpFilterTest.java

```diff
@@ -1,17 +1,78 @@
 package se.liu.ida.hefquin.engine.queryplan.executable.impl.ops;
 
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Iterator;
+
+import org.apache.jena.datatypes.xsd.XSDDatatype;
+import org.apache.jena.graph.Node;
+import org.apache.jena.graph.NodeFactory;
+import org.apache.jena.sparql.core.Var;
+import org.apache.jena.sparql.engine.binding.Binding;
+import org.apache.jena.sparql.expr.Expr;
+import org.apache.jena.sparql.util.ExprUtils;
 import org.junit.Test;
 
+import se.liu.ida.hefquin.engine.data.SolutionMapping;
+import se.liu.ida.hefquin.engine.data.utils.SolutionMappingUtils;
+import se.liu.ida.hefquin.engine.queryplan.executable.ExecOpExecutionException;
+import se.liu.ida.hefquin.engine.queryplan.executable.IntermediateResultElementSink;
+import se.liu.ida.hefquin.engine.queryplan.executable.impl.GenericIntermediateResultBlockImpl;
+import se.liu.ida.hefquin.engine.queryplan.executable.impl.pullbased.TestUtils;
+
 public class ExecOpFilterTest
 {
 	@Test
-	public void test1_RenameTheseTestMethodsToSomethingThatDescribesTheirPurpose() {
+	public void filter_Numbers() {
+		final FilterTestSink sink = new FilterTestSink();
+		final GenericIntermediateResultBlockImpl resultBlock = new GenericIntermediateResultBlockImpl();
+		final Expr lessThan10 = ExprUtils.parse("?x < 10");
+		
+		final Node value8 = NodeFactory.createLiteral("8", XSDDatatype.XSDinteger);
+		final Node value12 = NodeFactory.createLiteral("8", XSDDatatype.XSDinteger);
+		final Var x8 = Var.alloc("x");
+		final Var x12 = Var.alloc("x");
```

Comment:
There is no need to create two java objects that represent the same query variable. Simply do:
```java
final Var x = Var.alloc("x");
```
... and then use that program variable `x` wherever you are currently using (or planning to use) `x8` and `x12`.